client/termsetting/handlers/geneVariant.ts

```diff
@@ -105,23 +108,24 @@ export function fillTW(tw: GeneVariantTW, vocabApi: VocabApi, defaultQ: GeneVari
 
 	// groupsetting
 	// fill term.groupsetting
-	// !!! TODO: fix any type here, common.geneVariantTermGroupsetting should have the correct data shape or use a different type def !!!
-	if (!tw.term.groupsetting) tw.term.groupsetting = geneVariantTermGroupsetting //as any
+	if (!tw.term.groupsetting) tw.term.groupsetting = geneVariantTermGroupsetting
 	// fill q.groupsetting
-	if (tw.q.type == 'groupsetting') {
+	if (tw.q.type == 'predefined-groupset' || tw.q.type == 'custom-groupset') {
 		// groupsetting in use
-		// fill a single data type
+		// must specify a single data type
 		if (!tw.q.dt) {
 			const ds_dts = getDsDts(vocabApi.termdbConfig.queries)
 			tw.q.dt = ds_dts[0]
 		}
-		// fill a single orign, if applicable
+		// must specify a single orign (if dt has annotated origins)
 		const byOrigin = vocabApi.termdbConfig.assayAvailability?.byDt[tw.q.dt]?.byOrigin
 		if (byOrigin && !tw.q.origin) tw.q.origin = 'somatic'
 		// fill groupsetting
-		if (!tw.q.groupsetting.kind) {
-			const groupset_idxs = getGroupsetIdxs(tw.q.dt)
-			tw.q.groupsetting = { kind: 'predefined', predefined_groupset_idx: groupset_idxs[0] }
+		if (tw.q.type == 'predefined-groupset') {
+			if (!Number.isInteger(tw.q.predefined_groupset_idx)) {
+				const groupset_idxs = getGroupsetIdxs(tw.q.dt)
+				tw.q.predefined_groupset_idx = groupset_idxs[0]
+			}
 		}
 	}
 
```

Comment:
Curious about the decision to not use literal `defaultProps` object, and keep the multi-line/step-by-step adding of properties.

Reply:
Not sure how to avoid the step-by-step approach here, since the if-else statements are necessary to add properties based on certain conditions. What are your thoughts?

Reply:
we can optimize later in a separate branch, in the termwrapper refactor